src/NServiceBus.Transport.Msmq/NonDurable/ExpressMessageExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.Transport.Msmq
+{
+    using System;
+    using Configuration.AdvancedExtensibility;
+
+    /// <summary>
+    /// Provides extension methods to configure express messages.
+    /// </summary>
+    public static class ExpressMessageExtensions
+    {
+        /// <summary>
+        /// Sets the function to be used to evaluate whether a type is an express message or not.
+        /// </summary>
+        public static void DefiningExpressMessagesAs(this RoutingSettings<MsmqTransport> routingSettings, Func<Type, bool> expressMessageConvention)
```

Comment:
Related to my comment above, maybe we could change the type of the callback here to be of type `Func<TransportMessage, boo>` so that the application of the convention can happen in the transport?

Reply:
as you said yourself in the previous comment, this doesn't give us information about the message type unfortunately. Given this functionality is message-type convention based, this is kinda crucial